File: May-Leetcoding-Challenge/11. Flood Fill/FloodFill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) 
    {
        int color = image[sr][sc];
        int row = image.length, col = image[0].length;
        
        if(color == newColor)
            return image;
        
        dfs(image, sr, sc, color, newColor);
        return image;
    }
    public void dfs(int[][] image, int r, int c, int color, int newColor)
        {
            if(r < 0 || r >= image.length || c < 0 || c >= image[0].length || image[r][c] != color)
                return;
            
            image[r][c] = newColor;
            dfs(image, r+1, c, color, newColor);
            dfs(image, r-1, c, color, newColor);
            dfs(image, r, c+1, color, newColor);
            dfs(image, r, c-1, color, newColor);
        }
}

File: May-Leetcoding-Challenge/11. Flood Fill/FloodFill.py
class Solution:
    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        
        row, col = len(image), len(image[0])
        color = image[sr][sc]
        
        if color == newColor:
            return image
        
        def dfs(r, c):
                if r < 0 or r >= row or c < 0 or c >= col or image[r][c] != color:
                    return
                image[r][c] = newColor
                dfs(r-1, c)
                dfs(r+1, c)
                dfs(r, c-1)
                dfs(r, c+1)
                
        dfs(sr, sc)
        return image